### PR TITLE
[Not Modular]Blood type option in the character setup, more blood types

### PR DESCRIPTION
**Pull request by BobJoga**
## About The Pull Request

Adds a bunch of new bloodtypes for various species and a bloodtype option on the character setup.

## Why It's Good For The Game

Powergamer AB+ time

## Changelog
:cl:
add: Character setup now has blood type selection.
add: Many new blood types for various species.
/:cl:

### CODE REVIEW

**Comment by Strazyplus** [comment]
I was hoping this was merged by the time I returned from my short break.

**Comment by Strazyplus** [comment]
image

**Comment by BobJoga** [comment]
> image
> 
> sorry I had to
> Yes bob thank you for fixing bugs, i'd be mad if I could not nom powah.

the PR is getting stale and getting conflicts because of how damn long it's taking to be merged.
i'm just trying to fix merge conflicts.

**Comment by BobJoga** [comment]
fuck this shit

**Comment by Strazyplus** [comment]
REEEEEEEEEEEEEEEEEEEEEEEEEEEEEE

**Comment by Strazyplus** [comment]
Disaster adverted 
tenor

File: code/controllers/subsystem/processing/quirks.dm [hunk 1]
```diff
@@ -14,10 +14,17 @@ PROCESSING_SUBSYSTEM_DEF(quirks)
...
 
+	//SKYRAT CHANGE - Blood
+	var/list/all_bloodtypes = list()
+	//yes this is terrible but i want to make it automatic
+
 /datum/controller/subsystem/processing/quirks/Initialize(timeofday)
 	if(!quirks.len)
 		SetupQuirks()
 		quirk_blacklist = list(list("Blind","Nearsighted"),list("Jolly","Depression","Apathetic"),list("Ageusia","Deviant Tastes"),list("Ananas Affinity","Ananas Aversion"),list("Alcohol Tolerance","Alcohol Intolerance"),list("Alcohol Intolerance","Drunken Resilience"),list("Speech impediment (r as l)","Speech impediment (l as w)","Speech impediment (r as w)", "Speech impediment (r and l as w)")) // Skyrat edit
+	if(!all_bloodtypes.len)
```
Comment (Akrilla) [inline code/controllers/subsystem/processing/quirks.dm:25]: Skyrat change

File: code/modules/client/preferences.dm [hunk 2]
```diff
@@ -102,7 +102,8 @@ GLOBAL_LIST_INIT(food, list( // Skyrat addition
 	var/gender = MALE					//gender of character (well duh)
 	var/age = 30						//age of character
 	//SKYRAT CHANGES
-	var/skyrat_ooc_notes = ""
+	var/bloodtype = ""
+	var/skyrat_ooc_notes
```
Comment (Akrilla) [inline code/modules/client/preferences.dm:106]: same

File: code/modules/client/preferences.dm [hunk 3]
```diff
@@ -421,6 +422,9 @@ GLOBAL_LIST_INIT(food, list( // Skyrat addition
 			dat += 	"<a href='?_src_=prefs;preference=flavor_background;task=input'>Background</a>"
 			dat += 	"<a href='?_src_=prefs;preference=character_skills;task=input'>Skills</a><br>"
 			dat += 	"<a href='?_src_=prefs;preference=exploitable_info;task=input'>Exploitable Information</a><br>"
+			if(pref_species.bloodtypes.len)
+				dat += "Blood type :"
```
Comment (Akrilla) [inline code/modules/client/preferences.dm:426]: etc

File: code/modules/client/preferences.dm [hunk 4]
```diff
@@ -1868,6 +1872,14 @@ GLOBAL_LIST_INIT(food, list( // Skyrat addition
 					var/msg = stripped_multiline_input(usr, "Set your OOC Notes", "OOC Notes", skyrat_ooc_notes, MAX_FLAVOR_LEN, TRUE)
 					if(msg)
 						skyrat_ooc_notes = html_decode(msg)
+				
+				if("bloodtype")
+					var/msg = input(usr, "Choose your blood type", "Blood Type", "") as anything in (pref_species.bloodtypes + "Default")
+					if(msg)
```
Comment (Akrilla) [inline code/modules/client/preferences.dm:1878]: etc

File: code/modules/client/preferences.dm (hunk 2 above)
Comment (BobJoga) [reply code/modules/client/preferences.dm:106]: it's denoted above

File: code/modules/client/preferences.dm (hunk 4 above)
Comment (BobJoga) [reply code/modules/client/preferences.dm:1878]: it's in the skyrat change section its good

File: code/modules/client/preferences.dm (hunk 2 above)
Comment (Akrilla) [reply code/modules/client/preferences.dm:106]: mb

File: code/modules/client/preferences.dm (hunk 4 above)
Comment (Akrilla) [reply code/modules/client/preferences.dm:1878]: mb

**Pull request merged**